### PR TITLE
Add some syntax sugar for when backends use the same function.

### PR DESCRIPTION
**Pull request by ezyang**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #40469 Generalize Python dispatcher testing API; disallow overwriting fallback
* #40184 Delete requires_tensor
* **#40182 Add some syntax sugar for when backends use the same function.**

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Differential Revision: [D22187354](https://our.internmc.facebook.com/intern/diff/D22187354)

### CODE REVIEW

File: aten/src/ATen/native_parse.py [hunk 1]
```diff
@@ -419,7 +436,8 @@ def run(paths):
                 declaration['manual_kernel_registration'] = func.get('manual_kernel_registration', False)
                 declaration['category_override'] = func.get('category_override', '')
                 declaration['arguments'] = func.get('arguments', arguments)
-                declaration['type_method_definition_dispatch'] = func.get('dispatch', declaration['name'])
+                declaration['type_method_definition_dispatch'] = \
```
Comment (bhosmer) [inline aten/src/ATen/native_parse.py:439]: Just curious, does this allow silent shadowing of dupes that the old one would've caught? I don't know if the YAML parser errors on duplicate keys, but if so, we'd now allow {"CPU, Foo": ..., "CPU, BAR": ...} and silently pick a winner. 

I'm really not sure it's worth doing the extra cross-key dupe checking but seemed worth asking about.

**Review by bhosmer — APPROVED** [review]
❤️ 

One note inline, not sure it's worth acting on.

File: aten/src/ATen/native_parse.py (hunk 1 above)
Comment (ezyang) [reply aten/src/ATen/native_parse.py:439]: This is tested for in https://github.com/pytorch/pytorch/pull/40182/files#diff-24361820a647fae0fc49c38f7768aeb4R377

Testing for dupe shadowing caught a number of bugs in this diff!

Comment (bhosmer) [reply aten/src/ATen/native_parse.py:439]: Yeah saw that, but it tests for dupes within not across dispatch entries, right? I was thinking of the latter - imagining YAML might already be checking for e.g. 
```
  - CPU: ...
  - CPU: ...
```
but not 
```
  - CPU, Foo: ...
  - CPU, Bar: ...
```
But again I'm not sure that kind of user error will be common enough to complicate things to check for.

**Pull request closed without merging**